plugins/service/renderer/srv6/srv6_renderer.go

```diff
@@ -339,14 +378,29 @@ func (r *Renderer) renderService(service *renderer.ContivService, oper operation
 		// adding LocalSID
 		_, usedInOtherService := otherServiceLocalBackends[backend.Key()]
 		if !(oper == serviceDel && usedInOtherService) { // don't delete local sid if it is used in other service
-			localSID := &vpp_srv6.LocalSID{
-				Sid:               r.IPAM.SidForServicePodLocalsid(backend.ip).String(),
-				InstallationVrfId: r.ContivConf.GetRoutingConfig().PodVRFID,
-				EndFunction: &vpp_srv6.LocalSID_EndFunction_DX6{EndFunction_DX6: &vpp_srv6.LocalSID_EndDX6{
-					NextHop:           backend.ip.String(),
-					OutgoingInterface: vppIfName,
-				}},
+			var localSID *vpp_srv6.LocalSID
+			if isIPv6(r.ContivConf.GetIPAMConfig().PodSubnetCIDR.IP) {
```

Comment:
This should probably be r.ContivConf.GetIPAMConfig().UseIPv6. It is basically the same, but it handles some special cases better:
1. not filled PodSubnetCIRD defaulting to IPv4
2. pod subnet CIDR can be also defined in one big string (ContivCIDR) together with other subnet CIDRs (i never used this, but it is possible)

And also IPv6/IPv4 code switches can be tracked easily by finding usage of this config variable.

Reply:
fixed